flow/IThreadPool.h

```diff
@@ -92,12 +92,12 @@ class ThreadReturnPromise : NonCopyable {
 	void send( T const& t ) {  // Can be called safely from another thread.  Call send or sendError at most once.
 		Promise<Void> signal;
 		tagAndForward( &promise, t, signal.getFuture() );
-		g_network->onMainThread( std::move(signal), g_network->getCurrentTask() | 1 );
+		g_network->onMainThread( std::move(signal), incrementPriority( g_network->getCurrentTask() ) );
```

Comment:
Is there any scenario in which this behaves differently than before? Maybe as a way to check, we could assert (or assert_we_think) that the current task doesn't have the low bit set. I'm not sure how effectively the simulator tests priorities now, but if it does a decent job that could be useful.

Reply:
Oh, marking as resolved doesn't post the comment.  :(

Added an `incrementPriorityIfEven` to avoid doing two things at once.  The logic here seems strange to me though.

Reply:
Going back in history, it looks like the priority here was changed to be a `+1` when trying to solve some priority inversion related to coro stuff. It was later changed to `|1` to apparently solve an issue where priorities were being increased repeatedly:

> prevent threadReturnPromise from continually increasing priority

It looks a little goofy to me too and seems to have a weird dependency on the priorities in question being even, but we probably don't have to worry about that in this PR.